src/evict/evict_lru.c

```diff
@@ -2673,13 +2673,27 @@ __wt_verbose_dump_cache(WT_SESSION_IMPL *session)
 		    WT_DHANDLE_NEXT(session, dhandle, &conn->dhqh, q));
 		if (dhandle == NULL)
 			break;
+		/*
+		 * Lock the handle to prevent changes in the handle's state
+		 * when it is being used to dump the cache information.
+		 */
+		if ((ret = __wt_try_readlock(session, &dhandle->rwlock)) ==
+		    EBUSY)
+			continue;
+		WT_RET(ret);
```

Comment:
Since this code only exists for diagnostics, is there any problem with just wrapping the whole loop in `WT_WITH_HANDLE_LIST_READ_LOCK`?

Reply:
I think that this code can be invoked by the `debug_info` API too. This API is not documented but it can be used by the test programs. In that case, wrapping the loop with the `WT_WITH_HANDLE_LIST_READ_LOCK`, could that interfere with other operations?

Reply:
My concern with that is if there are cases where we want to dump the cache but can't get that lock. They are orthogonal, so this approach seems reasonable if I think about it.

Reply:
@nehakhatri5 can you answer your question?

Reply:
We actually do not want other threads to get a lock on the dhandle and change its state when dumping the cache. So it is the purpose rather than interference.